lib/logstash/inputs/file.rb

```diff
@@ -181,27 +182,47 @@ def register
     if @start_position == "beginning"
       @tail_config[:start_new_files_at] = :beginning
     end
+
+    if (@run_with_identity = @codec.class.config_name == 'multiline')
+      @codec = LogStash::Codecs::IdentityMapCodec.new(@codec)
+    end
```

Comment:
Actually, now that I have had time to reflect, I am going to be bold and wrap all codecs. It simplifies the block code and improves readability.